docassemble/assemblylinewizard/interview_generator.py

```diff
@@ -887,97 +891,140 @@ def get_docx_variables( text ):
+    if docassemble_var in reserved_whole_words:
+        return docassemble_var
+
+    if docassemble_var in reserved_var_plurals:
+        return docassemble_var + GATHER_CALL
 
-  # Break up label into its parts
-  label_groups = get_reserved_label_parts(reserved_prefixes, label)
+    # Everything before the first period and everything from the first period to the end
+    var_parts = re.findall(r'([^.]*)(\..*)*', docassemble_var)
 
-  # If no matches to automateable labels were found,
-  # just use the label as it is
-  if (label_groups is None or label_groups[1] == ''):
-    return label  # return without 
-    
-  # With reserved words, we're always using an index
-  # of the plural version of the prefix of the label
-  prefix = label_groups[1]
-  # Turn any singluars into plurals, e.g. 'user' into 'users'
-  var_start = pluralize_base(reserved_pluralizers_map, prefix)
-  if prefix == label: # it's just a standalone, like "defendant"
-    return var_start # Return the pluralized standalone variable
-
-  digit = label_groups[2]
-  index = indexify(digit)
-
-  # If it's a numbered singluar reserved prefix, e.g. user3
-  if label == prefix + digit:
-    # Return the plural plus the index, e.g. users[2]
-    return var_start + index
+    # test for existence (empty strings result in a tuple)
+    if not var_parts[0]:
+        return False
+    # The prefix, ensuring no key or index
+    prefix = re.sub(r'\[.+\]', '', var_parts[0][0])
+    has_reserved_prefix = prefix in reserved_pluralizers_map.values()
 
-  suffix = label_groups[3]
-  # Avoid transforming arbitrary suffixes into attributes
-  if not suffix in reserved_suffixes_map:
-    return label  # return it as is
-  else:    
-    suffix_as_attribute = turn_any_suffix_into_an_attribute(reserved_suffixes_map, suffix)
-    whole_var = "".join([var_start, index, suffix_as_attribute])
-
-    # Has to happen after docket number has a bracket
-    if (should_be_stringified(whole_var)):
-      result = get_stringifiable_version(whole_var)
-    else: result = whole_var
-
-    return result
-  
-
-def is_reserved_docx_label(label, docx_only_suffixes = generator_constants.DOCX_ONLY_SUFFIXES, reserved_whole_words = generator_constants.RESERVED_WHOLE_WORDS, reserved_pluralizers_map = generator_constants.RESERVED_PLURALIZERS_MAP, reserved_suffixes_map=generator_constants.RESERVED_SUFFIXES_MAP):
+    if not has_reserved_prefix:
+        return docassemble_var
+
+    suffix = var_parts[0][1]
+    if not suffix:  # If only the prefix
+        return prefix + GATHER_CALL
+    # If the suffix is also reserved
+    # Regex for finding all exact matches of docx suffixes
+    docx_only_suffixes_regex = '^' + '$|^'.join(docx_only_suffixes) + '$'
+    docx_suffixes_matches = re.findall(docx_only_suffixes_regex, suffix)
+    if (suffix in reserved_suffixes_map.values()
+        or len(docx_suffixes_matches) > 0):
+        # TODO(brycew): this should call `gather(complete_attribute='...')`
```

Comment:
I'm not sure about this. If I'm not misunderstanding then a couple things come to mind:
1. Isn't `complete_attribute` part of the object creation process, not part of gathering? `plaintiffs: Individual.using(complete_attribute="foo")` Sorry if I missed something in the docs.
2. If I'm understanding correctly then if we did this, a suffix of `.address.address` would create `.gather(complete_attribute="address.address")`.
3. If the above is true, those would each be unique and we would end with several different different `plaintiffs.gather(complete_attribute="foo")` items in the interview order.

Reply:
I'm not positive as I haven't used that syntax for .gather(), but we should not make .gather() trigger a unique complete attribute. It's built-in to the gather process already to use the complete_attribute specified in the object creation, or whatever is need to render the object to a string. It's built-in to the process for Individual and Address.

Reply:
1. It can be both I believe. https://docassemble.org/docs/objects.html#DAList.gather
2. Yep! That's what I wanted to avoid, since I don't think `address.address` would work. I didn't want to get bogged down in getting that exactly right, so I just added a TODO.
3. That's another reason I made this a TODO, I thought we should handle multiple attributes as well. It looks like the recommended way to do this:
```objects:
  - friend: |
      DAList.using(
        object_type=Individual,
        complete_attribute='complete',
        there_are_any=True)
---
...
---
code: |
  friend[i].name.first
  friend[i].birthdate
  friend[i].favorite_animal
  friend[i].complete = True
  ```
  So questions are asked until `.complete` is True. Which is definitely out of scope for this PR.

Reply:
@nonprofittechy That makes sense, I didn't know it was preferred to stick to the constructor `complete_attribute`, or that DA already gathered all of those attributes. I'll remove this TODO then.